### PR TITLE
Add `today` alias for `current_date`

### PR DESCRIPTION
**Pull request by smallzhongfeng**
## Which issue does this PR close?

Closes #.

## Rationale for this change

ck support udf `today` https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions#:~:text=Accepts%20zero%20arguments%20and%20returns%20the%20current%20date%20at%20one%20of%20the%20moments%20of%20query%20analysis.%20The%20same%20as%20%E2%80%98toDate(now())%E2%80%99.

duckdb https://duckdb.org/docs/archive/0.9.2/sql/functions/date#:~:text=1992%2D04%2D01-,today(),-Current%20date%20(start
## What changes are included in this PR?

Add a new name for `current_date`

### CODE REVIEW

**Review by comphead — COMMENTED** [review]
Thanks @smallzhongfeng for this PR
Please add also tests and user documentation `scalar_functions.md`

**Comment by comphead** [comment]
> > Thanks @smallzhongfeng for this PR Please add also tests and user documentation `scalar_functions.md`
> 
> Thanks. Updated.

Please add `today` to the function list as well, see `date_trunc`, `datetrunc` as an example.

Also the tests are needed, we can likely use `timestamps.slt` sqllogic file as the test location

**Comment by smallzhongfeng** [comment]
I add a test, because the function `today` is real-time, I can only compare it with the function `now`.

**Review by alamb — APPROVED** [review]
Thank you @smallzhongfeng

**Comment by comphead** [comment]
Thanks @smallzhongfeng 
Please add `today` to the function list in the doc, like in https://github.com/apache/arrow-datafusion/blob/9be75f71eb59a359674669f5c2688f49f44b4050/docs/source/user-guide/sql/scalar_functions.md?plain=1#L1282

File: datafusion/sqllogictest/test_files/timestamps.slt [hunk 1]
```diff
@@ -46,6 +46,25 @@ statement ok
...
+query B
+select now() = current_date();
+----
+false
+
+query B
+select cast(now() as date) = today();
+----
+true
+
```
[the remark this answers is not in the document]
Comment (smallzhongfeng) [reply datafusion/sqllogictest/test_files/timestamps.slt:72]: Updated~

**Review by alamb — APPROVED** [review]
Thank you again @smallzhongfeng  -- this looks great

File: datafusion/sqllogictest/test_files/timestamps.slt (hunk 1 above)
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/sqllogictest/test_files/timestamps.slt:72]: 🤔  this seems to fail on CI now

File: datafusion/sqllogictest/test_files/timestamps.slt [hunk 2]
```diff
@@ -46,6 +46,30 @@ statement ok
...
+query B
+select now() = current_date();
+----
+false
+
+query B
+select current_date() = today();
+----
+false
```
[the remark this answers is not in the document]
Comment (smallzhongfeng) [reply datafusion/sqllogictest/test_files/timestamps.slt:66]: My fault.

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Thank you @Weijun-H

**Pull request merged**